src/main/host/host.rs

```diff
@@ -80,107 +136,516 @@ impl Host {
     }
 
     pub fn id(&self) -> HostId {
-        unsafe { cshadow::host_getID(self.chost()) }
+        unsafe { cshadow::hostc_getID(self.chost()) }
     }
 
     pub fn name(&self) -> &str {
-        let slice = unsafe { std::ffi::CStr::from_ptr(cshadow::host_getName(self.chost())) };
+        let slice = unsafe { std::ffi::CStr::from_ptr(cshadow::hostc_getName(self.chost())) };
         slice.to_str().unwrap()
     }
 
     pub fn default_ip(&self) -> std::net::IpAddr {
         use std::net;
-        let addr = unsafe { cshadow::host_getDefaultIP(self.chost()) };
+        let addr = unsafe { cshadow::hostc_getDefaultIP(self.chost()) };
         net::IpAddr::V4(net::Ipv4Addr::from(addr.to_le_bytes()))
     }
 
     pub fn abstract_unix_namespace(&self) -> &Arc<AtomicRefCell<AbstractUnixNamespace>> {
-        let ptr = unsafe { cshadow::host_getAbstractUnixNamespace(self.chost()) };
+        let ptr = unsafe { cshadow::hostc_getAbstractUnixNamespace(self.chost()) };
         assert!(!ptr.is_null());
         unsafe { &*ptr }
     }
 
     pub fn log_level(&self) -> Option<log::LevelFilter> {
-        let level = unsafe { cshadow::host_getLogLevel(self.chost()) };
+        let level = unsafe { cshadow::hostc_getLogLevel(self.chost()) };
         crate::core::logger::log_wrapper::c_to_rust_log_level(level).map(|l| l.to_level_filter())
     }
 
-    pub fn random(&mut self) -> &mut impl rand::Rng {
-        let ptr = unsafe { cshadow::host_getRandom(self.chost()) };
+    pub fn random(&self) -> &mut impl rand::Rng {
+        let ptr = unsafe { cshadow::hostc_getRandom(self.chost()) };
         let random = unsafe { ptr.as_mut() }.unwrap();
         &mut random.0
     }
```

Comment:
Sorry for the last-minute comment. I don't think this function is safe, since it's returning a mutable reference without any kind of interior mutability. For example, the following code would cause two mutable references to the same data:

```rust
let rng_1 = host.random();
let rng_2 = host.random();
```

Reply:
Oops, good catch. I'll probably move the Rng into the Rust Host and wrap it in a RootedRefCell